e2e/common/traits/route_test.go

```diff
@@ -0,0 +1,382 @@
+	"github.com/apache/camel-k/pkg/util/openshift"
+)
+
+const(
+	secretName = "test-certificate"
+	servingCertificateSecret = "serving-certificate"
+	integrationName = "platform-http-server"
+	waitBeforeHttpRequest = 6 * time.Second
+)
+
+type keyCertificatePair struct {
+	Key []byte
+	Certificate []byte
+}
+
+// self-signed certificate used when making the https request
+var certPem []byte
+
+// the e2e tests run on openshift3 in CI and there is no object to retrieve the base domain name
+// to create a valid hostname to later have it validate when doing the https request
+// as this a e2e test to validate the route object and not the certificate itself 
+// if the base domain name cannot be retrieved from dns/cluster we can safely skip TLS verification
+// if running on openshift4, there is the dns/cluster object to retrieve the base domain name, 
+// then in this case the http client validates the TLS certificate
+var skipClientTLSVerification = false
+
+func TestRunRoutes(t *testing.T) {
+	WithNewTestNamespace(t, func(ns string) {
+		ocp, err := openshift.IsOpenShift(TestClient())
+		if !ocp {
+			t.Skip("This test requires route object which is available on OpenShift only.")
+			return
+		}
+		assert.Nil(t, err)
+
+		Expect(Kamel("install", "-n", ns).Execute()).To(Succeed())
+
+		// create a test secret of type tls with certificates
+		// this secret is used to setupt the route TLS object across diferent tests
+		secret, err := createSecret(ns)
+		assert.Nil(t, err)
+
+		// they refer to the certificates create in the secret and are reused the different tests
+		refKey := secretName + "/" + corev1.TLSPrivateKeyKey
+		refCert := secretName + "/" + corev1.TLSCertKey
+
+		// =============================
+		// Insecure Route / No TLS
+		// =============================
+		t.Run("Route unsecure http works", func(t *testing.T) {
+			Expect(Kamel("run", "-n", ns, "files/PlatformHttpServer.java").Execute()).To(Succeed())
+			Eventually(IntegrationPodPhase(ns, integrationName), TestTimeoutMedium).Should(Equal(corev1.PodRunning))
+			route := Route(ns, integrationName)
+			Eventually(route, TestTimeoutMedium).ShouldNot(BeNil())
+			// must wait a little time after route is created, before doing an http request, 
+			// otherwise the route is unavailable and the http request will fail
+			time.Sleep(waitBeforeHttpRequest)
+			url := fmt.Sprintf("http://%s/hello?name=Simple", route().Spec.Host)
+			response, err := httpRequest(t, url, false)
+			assert.Nil(t, err)
+			assert.Equal(t, "Hello Simple", response)
+			Expect(Kamel("delete", "--all", "-n", ns).Execute()).Should(BeNil())
+		})
+
+		// =============================
+		// TLS Route Edge
+		// =============================
+
+		t.Run("Route Edge https works", func(t *testing.T) {
+			Expect(Kamel("run", "-n", ns, "files/PlatformHttpServer.java",
+				"-t", "route.tls-termination=edge",
+				"-t", "route.tls-certificate-secret=" + refCert,
+				"-t", "route.tls-key-secret=" + refKey,
```

Comment:
Why is a secret is needed with `edge` termination?

Reply:
To use `edge` termination a certificate is required and it comes from the `secret`.

Reply:
My understanding is that a certificate is not require for `edge` termination. The router’s default certificate is used for TLS termination, which is from my experience what users tend to rely on.

Reply:
I see the certificate is not required for `edge`, I can remove it, however as this is an e2e test it was my understanding to have the tests to cover all scenarios. I just tested locally with no certificate for edge and it works. Well, I can remove the certificate for edge ssl setup, if that's the case.

Reply:
One option is to keep that test as is, and add another one covering the scenario of `edge` termination with the router's default certificate, so we cover the most frequent scenario as well.

Reply:
Added another test route of type `edge` with default certificate.